src/docfx/lib/git/GitCommitProvider.cs

```diff
@@ -40,30 +38,34 @@ internal sealed class GitCommitProvider : IDisposable
         // Only the last N = MaxCommitCacheCountPerFile commit histories are cached for a file, they are selected by least recently used order (lruOrder).
         private readonly ConcurrentDictionary<string, Dictionary<(long commit, long blob), (long[] commitHistory, int lruOrder)>> _commitCache;
 
+        private static readonly ConcurrentDictionary<string, GitCommitProvider> s_gitCommitProvider = new ConcurrentDictionary<string, GitCommitProvider>();
```

Comment:
same as `RepositoryUtility`, static cache may contain stall data, use a `RepositoryProvider` instance whose lifetime is bound to a single build run.

Reply:
ok

---
In reply to: [237187074](https://github.com/dotnet/docfx/pull/3677#discussion_r237187074) [](ancestors = 237187074)

Reply:
so every watch build will reload the git commits again, interesting

Reply:
It right now is. There are a lot of state that could be shared across watch builds, I haven't started on that because we didn't have watch yet.

Reply:
so, why did you comment on that if we haven't got the watch design yet?